blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/QueueItemImpl.java

```diff
@@ -55,6 +57,15 @@ public int getExpectedBuildNumber() {
     }
 
     @Override
+    public void delete(){
+        if(!item.hasCancelPermission()){
+            throw new ServiceException.ForbiddenException(String.format("Not authorized to stop queue: %s", getId()));
+        }
+
+        Jenkins.getInstance().getQueue().cancel(item);
```

Comment:
@michaelneale Depends on the authorization strategy. For example admin can set a user to only cancel specific projects. Basically authorization is applicable at any item level, be it job or folder or run and it could be applicable system wide or per user or a group.